### PR TITLE
Set executor hostname via cloud-init

### PR DESCRIPTION
**Pull request by croissanne**
This way much more of the journal will be captured under the new hostname.

### CODE REVIEW

File: templates/packer/ansible/roles/common/files/worker-initialization-scripts/set_executor_hostname.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -euo pipefail
```
Comment (mvo5) [inline templates/packer/ansible/roles/common/files/worker-initialization-scripts/set_executor_hostname.sh:2]: (slightly unrelated) Fwiw, `pipefail` has it's own set of weirdness , see https://mvogt.wordpress.com/2018/07/23/bash-o-pipefail-can-have-pitfailswpitfalls/ - not that it matters here.

File: internal/cloud/awscloud/secure-instance_test.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 func TestSecureInstanceUserData(t *testing.T) {
 	type testCase struct {
 		CloudWatchGroup  string
+		Hostname         string
```
Comment (mvo5) [inline internal/cloud/awscloud/secure-instance_test.go:11]: :heart:

**Review by mvo5 — APPROVED** [review]
Thank you!

File: templates/packer/ansible/roles/common/files/worker-initialization-scripts/set_executor_hostname.sh (hunk 1 above)
Comment (croissanne) [reply templates/packer/ansible/roles/common/files/worker-initialization-scripts/set_executor_hostname.sh:2]: oh interesting, had no idea

**Pull request merged**